src/main/scala/com/banzaicloud/spark/metrics/sink/PrometheusSink.scala

```diff
@@ -279,9 +297,31 @@ abstract class PrometheusSink(property: Properties,
     }
   }
 
+  /**
+    * 2 sources for labels:
+    *  1. from SparkConf
+    *  2. property labels
+    * @return
+    */
+  private def collectLabels(sparkConf: SparkConf): Map[String, String] = {
+    val configLabels = Option(property.getProperty(KEY_LABELS))
+        .flatMap(parseLabels).getOrElse(Map.empty)
+
+    val sparkConfLabels = collectLabelsFromSparkConf(sparkConf)
+    configLabels ++ sparkConfLabels
+  }
+
+  private def collectLabelsFromSparkConf(sparkConf: SparkConf): Map[String, String] = {
```

Comment:
```suggestion
  private def collectLabelsFromSparkConf(sparkConf: SparkConf): Try[Map[String, String]] = {
```

As parseLabel, may throw an exception.